File: Leetcode/49-Group-Anagrams/GroupAnagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    let hash = {}
    strs.forEach(str => {
        let letters = str.split('').sort()

        hash[letters] ? hash[letters].push(str) : hash[letters] = [str]
    })

    // const keys = Object.keys(hash)
    // const values = keys.map(function(v) {return hash[v]})
    const values = Object.values(hash)
    return values
};

File: Leetcode/49-Group-Anagrams/GroupAnagrams.go
func groupAnagrams(strs []string) [][]string {
    m := make(map[string][]string, 0)
    res := make([][]string, 0)
    for _, val := range strs {
        strArr := strings.Split(val, "")
        sort.Strings(strArr)
        sortedStr := strings.Join(strArr, "")
        m[sortedStr] = append(m[sortedStr], val)
    }
    for _, val := range m {
        res = append(res, val)
    }
    return res
}
